File: sdk/typescript/src/types/name-service.ts
// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

import type { Infer } from 'superstruct';
import { array, boolean, nullable, object, string } from 'superstruct';

export const ResolvedNameServiceNames = object({
	data: array(string()),
	hasNextPage: boolean(),
	nextCursor: nullable(string()),
});
export type ResolvedNameServiceNames = Infer<typeof ResolvedNameServiceNames>;

File: crates/sui-graphql-rpc/src/types/name_service.rs
// Copyright (c) Mysten Labs, Inc.
// SPDX-License-Identifier: Apache-2.0

use async_graphql::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
pub(crate) struct NameService(pub String);
scalar!(NameService, "NameService");
